### DISCUSSION TITLE
Special @types/node handling

### DISCUSSION THREAD

**Discussion by rarkins** [renovatebot/renovate#21662] [Suggest an Idea]
We should try to detect the version of `node` in use and not upgrade past it.

**Comment by rarkins** [comment]
For now, the default preset will disable major updates for `@types/node`

**Comment by ghost** [comment]
This same issue was brought to my attention recently, and your change to the standard Renovate config has been helpful to us in addressing the odd synchronization issues between a `node` engine and the `@types/node` package.

Would it make sense to synchronize upgrades of `@types/node` with the upgrading of the `node` engine property in `package.json`?

**Comment by rarkins** [comment]
Waiting on getPackageUpdates refactor

**Comment by HonkingGoose** [comment]
What's the status of this issue now? 

`getPackageUpdates` was refactored with PR #1994, would that get this issue unblocked, or is further work needed elsewhere?

**Comment by keeganwitt** [comment]
#519 seems similar/related, since it also discusses coupled packages (e.g. jest and @types/jest). Though this issue might have additional considerations, since it's dealing with platform upgrades.

**Comment by karlhorky** [comment]
> For now, the default preset will disable major updates for `@types/node`

Where is this configured? Is it [`helpers:disableTypesNodeMajor`](https://docs.renovatebot.com/presets-helpers/#helpersdisabletypesnodemajor)?

Would the following configuration re-enable major upgrades of `@types/node`?

**Update:** Nope, that doesn't do it... 🤔

```json
{
  "extends": [
    "config:base"
  ],
  "packageRules": [
    {
      "matchPackageNames": [
        "@types/node"
      ],
      "matchUpdateTypes": [
        "major"
      ],
      "enabled": true
    }
  ]
}
```

**Comment by karlhorky** [comment]
Ah it appears that there is also [`workarounds:typesNodeVersioning`](https://docs.renovatebot.com/presets-workarounds/#workaroundstypesnodeversioning), which appears to be the one that's enabled in the base Renovate config.

Seems to work to disable it using:

```json
{
  "extends": [
    "config:base"
  ],
  "packageRules": [
    {
      "description": "Enable major version upgrades of @types/node https://github.com/renovatebot/renovate/issues/1463#issuecomment-875014401",
      "matchManagers": [
        "npm"
      ],
      "matchPackageNames": [
        "@types/node"
      ],
      "versioning": "npm"
    }
  ]
}
```

**Update:** as @viceice mentions below, `ignorePresets` is a simpler option:

```json
{
  "extends": [
    "config:base"
  ],
  "ignorePresets": ["workarounds:typesNodeVersioning"]
}
```

**Comment by viceice** [comment]
I'm not sure, but `helpers:disableTypesNodeMajor` is not added to `:base` preset.

`workarounds:typesNodeVersioning` is included by `:base` preset and forces `node` versioning, which only accepts lts versions as stable, so you currently don't get an update from v14.x to any later major until v16 becomes lts around `2021-10-26`

https://nodejs.org/en/about/releases/

So it works as intended, as most users like to use only node lts versions.

**Comment by karlhorky** [comment]
Nice, cool to hear that it works for many people.

I think there's a bunch of people who like to stay on latest though, so [the post above](https://github.com/renovatebot/renovate/issues/1463#issuecomment-875014401) is probably useful for them, to disable this behavior of the base preset.

**Comment by viceice** [comment]
Using `ignorePresets`  is a better solution 😉 
```json
{
  "ignorePresets": ["workarounds:typesNodeVersioning"]
}
```

https://docs.renovatebot.com/configuration-options/#ignorepresets

**Comment by karlhorky** [comment]
Oh, that's even better, thanks! I'll update my post above too.

**Comment by rarkins** [comment]
Better: use node versioning for this
